packages/flutter_tools/lib/src/ios/xcode_debug.dart

```diff
@@ -411,6 +413,49 @@ class XcodeDebug {
       verboseLogging: verboseLogging,
     );
   }
+
+  /// Ensure the Xcode project is set up to launch an LLDB debugger. If these
+  /// settings are not set, the launch will fail with a "Cannot create a
+  /// FlutterEngine instance in debug mode without Flutter tooling or Xcode."
+  /// error message. These settings should be correct by default, but some users
+  /// reported them not being so after upgrading to Xcode 15.
+  void ensureXcodeDebuggerLaunchAction(File schemeFile) {
+    if (!schemeFile.existsSync()) {
+      _logger.printError('Failed to find ${schemeFile.path}');
```

Comment:
I've purposely set this error and some of the ones below to be non-breaking and just print out the error. It only throws if it successfully finds the scheme file and parse it and finds the settings are incorrect.

Reason for this is I plan to CP this into stable and although I'm pretty confident that none of these should error, there may be cases I'm unaware of. And by default, most projects should have these settings be correct. I don't want to prevent a valid project from running.

Let me know if you think I'm being too cautious 😅

Reply:
do you plan to mark it a failure AFTER the CP? or would you like it to be printError indefinitely?

Reply:
in terms of cherry-picking, I would say this is an appropriate level of caution :)